pallets/data-preservers/src/lib.rs

```diff
@@ -168,143 +318,53 @@ pub mod pallet {
-    }
-
-    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-    #[derive(
-        RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, Encode, Decode, CloneNoBound, TypeInfo,
-    )]
-    #[scale_info(skip_type_params(T))]
-    pub enum ParaIdsFilter<T: Config> {
-        AnyParaId,
-        Whitelist(BoundedVec<ParaId, T::MaxParaIdsVecLen>),
-        Blacklist(BoundedVec<ParaId, T::MaxParaIdsVecLen>),
-    }
-
-    impl<T: Config> ParaIdsFilter<T> {
-        pub fn len(&self) -> usize {
-            match self {
-                Self::AnyParaId => 0,
-                Self::Whitelist(list) | Self::Blacklist(list) => list.len(),
-            }
-        }
-    }
-
-    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-    #[derive(RuntimeDebug, PartialEq, Eq, Encode, Decode, Clone, TypeInfo)]
-    pub enum ProfileMode {
-        Bootnode,
-        Rpc { supports_ethereum_rpcs: bool },
-    }
-
-    /// Profile with additional data:
-    /// - the account id which created (and manage) the profile
-    /// - the amount deposited to register the profile
-    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
-    #[derive(
-        RuntimeDebugNoBound, PartialEqNoBound, EqNoBound, Encode, Decode, CloneNoBound, TypeInfo,
-    )]
-    #[scale_info(skip_type_params(T))]
-    pub struct RegisteredProfile<T: Config> {
-        pub account: T::AccountId,
-        pub deposit: BalanceOf<T>,
-        pub profile: Profile<T>,
-    }
-
-    /// Computes the deposit cost of a profile.
-    pub trait ProfileDeposit<Profile, Balance> {
-        fn profile_deposit(profile: &Profile) -> Result<Balance, DispatchErrorWithPostInfo>;
-    }
-
-    /// Implementation of `ProfileDeposit` based on the size of the SCALE-encoding.
-    pub struct BytesProfileDeposit<BaseCost, ByteCost>(PhantomData<(BaseCost, ByteCost)>);
-
-    impl<Profile, Balance, BaseCost, ByteCost> ProfileDeposit<Profile, Balance>
-        for BytesProfileDeposit<BaseCost, ByteCost>
-    where
-        BaseCost: Get<Balance>,
-        ByteCost: Get<Balance>,
-        Profile: Encode,
-        Balance: TryFrom<usize> + CheckedAdd + CheckedMul,
-    {
-        fn profile_deposit(profile: &Profile) -> Result<Balance, DispatchErrorWithPostInfo> {
-            let base = BaseCost::get();
-            let byte = ByteCost::get();
-            let size: Balance = profile
-                .encoded_size()
-                .try_into()
-                .map_err(|_| ArithmeticError::Overflow)?;
-
-            let deposit = byte
-                .checked_mul(&size)
-                .ok_or(ArithmeticError::Overflow)?
-                .checked_add(&base)
-                .ok_or(ArithmeticError::Overflow)?;
-
-            Ok(deposit)
-        }
-    }
+    #[pallet::storage]
+    #[pallet::getter(fn assignments)]
+    pub type Assignments<T: Config> = StorageMap<
+        _,
+        Blake2_128Concat,
+        ParaId,
+        BoundedVec<T::ProfileId, T::MaxAssignmentsPerParaId>,
```

Comment:
Also here
```suggestion
        BoundedBTreeSet<T::ProfileId, T::MaxAssignmentsPerParaId>,
```

Reply:
why this?

Reply:
To speed up contains, also to make it easier to insert and delete items. Performance-wise shouldn't matter it we keep the number of para ids small, so I would do the one that has better ux. Which I believe is the set, but if I'm wrong then it's fine.

Reply:
ALright I am fine with whatever, but I expect maybe oin ther order of 50-ish assignments per para max